src/app/loadout/loadout-utils.ts

```diff
@@ -63,6 +67,27 @@ export function getLight(store: DimStore, items: DimItem[]): number {
   }
 }
 
+/** Returns a map of armor hashes to stats. There should be just one of each item */
+export function getArmorStats(
+  defs: D1ManifestDefinitions | D2ManifestDefinitions,
+  items: DimItem[]
+): { [hashHash: number]: DimCharacterStat } {
+  const statsByArmorHash = armorStats
+    .map((statHash) => defs.Stat.get(statHash))
+    .reduce((ret, { hash, displayProperties: { description, icon, name } }) => {
+      ret[hash] = { hash, description, icon: bungieNetPath(icon), name, value: 0 };
+      return ret;
+    }, {});
```

Comment:
I'll admit, I almost always prefer a normal for loop to `reduce`.

Reply:
👍 updated to improve clarity